### ISSUE TITLE
Broken custom attributes rendering on Shipping Address renderer

### ISSUE THREAD

**Issue by iget-master** [magento/magento2#24490]
### Preconditions (*)

1. Using Magento 2.3.2
2.Have a Custom Attribute to Address as suggested by https://devdocs.magento.com/guides/v2.3/howdoi/checkout/checkout_new_field.html

### Steps to reproduce (*)
1. Go to checkout, create an address with the custom attribute filled
2. See the rendered address on address list

### Expected result (*)
1. It should print the custom attribute as it used to do on prior versions of Magento (2.2.x)

### Actual result (*)
It prints the attribute_code and value, but the attribute_code is an internal code, like `building_number`

### Details:

After v2.3.0, the `address().customAttributes` changed it data structure from:

```
customAttributes: {
    my_attribute_code: "attribute value"
}
```

to

```
customAttributes: [
    { attribute_code: "my_attribute_code", value: "attribute value" }
]
```

BUT: the address renderer didn't changed correctly, it looks like:

    <each args="data: address().customAttributes, as: 'element'">
        <each args="data: Object.keys(element), as: 'attribute'">
            <if args="typeof element[attribute] === 'object'">
                <if args="element[attribute].label">
                    <text args="element[attribute].label"/>
                </if>
                <ifnot args="element[attribute].label">
                    <if args="element[attribute].value">
                        <text args="element[attribute].value"/>
                    </if>
                </ifnot>
            </if>
            <if args="typeof element[attribute] === 'string'">
                <text args="element[attribute]"/>
            </if><br/>
        </each>
    </each>

### Suggested fix

The template should look like something like this:

    <each args="data: address().customAttributes, as: 'element'">
            <if args="typeof element['value'] === 'string'">
                <text args="element['value']"/>
            </if><br/>
        </each>
    </each>

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @dmytro-ch 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19937` were created

**Issue Available**: @dmytro-ch, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by ghost** [comment]
@magento I am working on it.

**Comment by magento-engcom-team** [comment]
Hi @iget-master. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24070` was closed as `non-reproducible` in `2.4-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem please update or provide additional information/steps/preconditions in the **_Description section_** and reopen this issue.
